### PR TITLE
Minor improvement for api

### PR DESCRIPTION
**Pull request by nicl-nno**
- Fixed bug in TS forecasting with API (data type was applied incorrectly)
- Added custom metric selection
- Bug with predict_proba fixed
-  tf version for old python is updated.
- compability with automlbenchmark (see https://github.com/nicl-nno/automlbenchmark/blob/test/frameworks/FEDOT/exec.py)

### CODE REVIEW

File: test/unit/api/test_main_api.py [hunk 1]
```diff
@@ -89,6 +88,20 @@ def test_api_forecast_correct(task_type: str = 'ts_forecasting'):
     assert metric['rmse'] >= 0
 
 
+def test_api_forecast_other_data_format_correct(task_type: str = 'ts_forecasting'):
```
Comment (J3FALL) [inline test/unit/api/test_main_api.py:91]: other_data_format - название неконкретное, что значит "другой" формат?

Comment (nicl-nno) [reply test/unit/api/test_main_api.py]: [Поменял](https://github.com/nccr-itmo/FEDOT/pull/251/files#diff-3024e99987ea3223b4f92729b426eabaf4b75077e6d3c3a550fde20d3a4e23fbR91) на test_api_forecast_numpy_input_with_static_model_correct

**Review by J3FALL — APPROVED** [review]

**Comment by nicl-nno** [comment]
Сделал ещё несколько обновлений для совместимости с automlbenchmark

File: fedot/api/api_utils.py [hunk 2]
```diff
@@ -13,18 +13,27 @@
...
+    auc=ClassificationMetricsEnum.ROCAUC,
+    f1=ClassificationMetricsEnum.f1,
+    logloss=ClassificationMetricsEnum.logloss,
+    mae=RegressionMetricsEnum.MAE,
+    mse=RegressionMetricsEnum.MSE,
+    msle=RegressionMetricsEnum.MSLE,
+    r2=RegressionMetricsEnum.R2,
+    rmse=RegressionMetricsEnum.RMSE,
+    silhouette=ClusteringMetricsEnum.silhouette
+)
```
Comment (J3FALL) [inline fedot/api/api_utils.py:29]: Я так и не понимаю, в чем польза инициализации словаря через dict, а не {} ?

File: fedot/api/api_utils.py [hunk 3]
```diff
@@ -70,11 +79,19 @@ def compose_fedot_model(train_data: InputData,
...
+    if metric is None:
+        metric_function = MetricByTask(task.task_type).metric_cls.get_value
+    else:
+        metric_id = _metrics_mapping.get(metric, None)
+        if metric_id is None:
+            raise ValueError(f'Incorrect metric {metric}')
+        metric_function = MetricsRepository().metric_by_id(metric_id)
 
+    # TODO search for best implementation ideas
     is_tuning = '_tun' in preset or preset == 'full'
```
Comment (J3FALL) [inline fedot/api/api_utils.py:95]: Получилось слишком компактно, что даже и не понятно, что имеется в виду.
Вообще, как я понял по тестам, preset - это набор используемых моделей + ты добавил еще постфикс _tun для проверки, нужен ли еще тюнинг или нет?
Мне кажется, экономия одного внешнего параметра того не стоит, ну будет with_tuning=False в функции compose_fedot_model - ничего такого вроде.
А так с этими префиксами выглядит не очень, ибо всегда можно забыть, какой там префикс должен быть.
Т.е. тут либо:
- Выделять спец. класс preset с двумя полями
- Делать 2 параметра - preset + with_tuning
- Делать "словарь" со всеми возможными комбинациями: ultra_light, light, light_tun, ultra_light_tun, и т.д.

File: fedot/api/main.py [hunk 4]
```diff
@@ -310,18 +316,20 @@ def get_metrics(self,
...
-        __metric_dict = {'rmse': RmseMetric.metric,
-                         'mae': MaeMetric.metric,
-                         'roc_auc': RocAucMetric.metric,
-                         'f1': F1Metric.metric,
-                         'silhouette': NotImplemented
+        __metric_dict = {'rmse': RMSE,
+                         'mae': MAE,
+                         'roc_auc': ROCAUC,
+                         'f1': F1,
+                         'silhouette': Silhouette
```
Comment (J3FALL) [inline fedot/api/main.py:332]: Немного смущает, что в скрипте выше есть словарь _metrics_mapping, в котором контент дублируется частично.
Так и задумывалось?

File: fedot/utilities/define_metric_by_task.py [hunk 5]
```diff
@@ -1,22 +1,22 @@
...
 class MetricByTask:
-    __metric_by_task = {TaskTypesEnum.regression: RmseMetric.metric,
-                        TaskTypesEnum.classification: RocAucMetric.metric,
-                        TaskTypesEnum.clustering: None,
-                        TaskTypesEnum.ts_forecasting: RmseMetric.metric,
+    __metric_by_task = {TaskTypesEnum.regression: RMSE,
+                        TaskTypesEnum.classification: ROCAUC,
+                        TaskTypesEnum.clustering: Silhouette,
+                        TaskTypesEnum.ts_forecasting: RMSE,
                         }
```
Comment (J3FALL) [inline fedot/utilities/define_metric_by_task.py:11]: Еще немного словарей с метриками ))

File: test/integration/quality/test_quality_improvement.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+    baseline_model.predict_proba(features=test_data_path)
+    baseline_metrics = baseline_model.get_metrics()
+
+    auto_model = Fedot(problem=problem, seed=42)
+    auto_model.fit(features=train_data_path, target='target')
+    auto_model.predict_proba(features=test_data_path)
+    auto_metrics = auto_model.get_metrics()
+
+    assert auto_metrics['roc_auc'] > baseline_metrics['roc_auc'] >= 0.827
```
Comment (J3FALL) [inline test/integration/quality/test_quality_improvement.py:28]: Почему именно 0.827 ?)

**Review by J3FALL — CHANGES_REQUESTED** [review]

File: test/integration/quality/test_quality_improvement.py (hunk 6 above)
Comment (nicl-nno) [reply test/integration/quality/test_quality_improvement.py]: Ну вот именно такое у бейзлайна качество на этих данных. Если оно снизилось - значит что-то сломано. [Вынес](https://github.com/nccr-itmo/FEDOT/pull/251/files#diff-c6b4340997ef324bb8c9df3307fd583cabefe52d225b0ee10c045df4eaa1d2f4R18) в переменную.

File: fedot/api/main.py (hunk 4 above)
Comment (nicl-nno) [reply fedot/api/main.py]: В идеале это разные метрики - одни для композирования, другие для оценки результата. Но в текущей реализации действительно разумно [убрать](https://github.com/nccr-itmo/FEDOT/pull/251/files#diff-c8c3532da680ee4d8a1c35a5b02009326da1b007a9fa794413345ffc864411eeR338) дублирование.

File: fedot/api/api_utils.py (hunk 2 above)
Comment (nicl-nno) [reply fedot/api/api_utils.py]: Ну, например меньше кавычек можно писать. Ок, поменял на {}

File: fedot/utilities/define_metric_by_task.py (hunk 5 above)
Comment (nicl-nno) [reply fedot/utilities/define_metric_by_task.py:11]: Ну этот и в SA используется, как раз и сослался на него, чтобы убрать дублирование.

File: fedot/api/api_utils.py (hunk 3 above)
Comment (nicl-nno) [reply fedot/api/api_utils.py]: В функции compose_fedot_model-то не проблема это сделать, но пока не придумал, как это удобно вывести наружу API. Вроде пользователю вообще не особо важно, происходит внутри тюнинг или нет. Мб тут что-то в стиле билдера нужно?

Пока вынес [сюда](https://github.com/nccr-itmo/FEDOT/pull/251/files#diff-c8c3532da680ee4d8a1c35a5b02009326da1b007a9fa794413345ffc864411eeR118-R126).

Comment (J3FALL) [reply fedot/api/api_utils.py]: Я не вижу ничего страшного, если в конструкторе Fedot будет опциональное поле - with_tuning.
Ну или если ты его сейчас в composer_params упрятал, но неплохо было бы в docstring или где-то еще раскрыть - что может быть передано в словаре composer_params

Comment (nicl-nno) [reply fedot/api/api_utils.py]: [Done](https://github.com/nccr-itmo/FEDOT/pull/251/files#diff-c8c3532da680ee4d8a1c35a5b02009326da1b007a9fa794413345ffc864411eeR57-R64)

Пример тоже добавил.

**Review by J3FALL — APPROVED** [review]
